### DISCUSSION TITLE
Machine Learning Models

### DISCUSSION THREAD

**Discussion by datatalking** [sfu-db/dataprep#911] [Q&A]
It notes in the February 3rd meeting notes that @zshandy is "Researching on SQL-fixing machine learning models @zshandy, also that there was an "Update dataprep.ai" processed.

So the url is for noting that the goal of the tool is to process it for Ai data cleaning but there is no Ai provided by the `dataprep` tool but I have a few questions about the interface or usage.

1. Seems like this runs headless through the terminal. oK
2. Seems like there are multiple 'dataprep demo.ipynb' notebooks
3. Seems like this isn't something that can run on its own... it needs human in the loop ?
4. What options are there to run based on a script I run, say pull data from ~/sandbox/dirty_data/datafile.txt or csv then it ouputs a clean file to a csv or a database ?
5. When I click the `configuration files` in the blue link below from the 'https://github.com/sfu-db/dataprep/blob/develop/examples/DataConnector_DBLP.ipynb [files`]([https://github.com/sfu-db/DataConnectorConfigs%3E]))
<img width="1226" alt="Screen Shot 2022-06-20 at 12 49 16 PM" src="https://user-images.githubusercontent.com/11813209/174669317-30aaea9d-81d2-4863-8fb4-38bb43854685.png">

How do most people use this, do you somehow memorize the 100+ funcitons or have them written down somewhere. This tool looks really good and might even replace the one I have built for myself but I'd like to upgrade the usage or onboarding part of it, the Videos are great, I'm just thinking how can I use this as part of my existing data process. Is there someone I could chat with via discord/slack etc or an indepth session as I work on collaborating?

**Comment by qidanrui** [comment]
Hi @datatalking,
   Thank you for being interested in our clean interface and watching the video! For your questions:
   1. Actually we run a small Flask server in your local machine as backend when running our clean interface
   2. Thanks for your suggestion, we will eliminate the redundancy
   3. It is human in the loop to fit for different users' needs
   4. Currently we support inputting Dataframe in your jupyter notebook and import csv from your local machine
   5. I think @wangxiaoying can answer your question 
   
Actually, we have our 100+ functions written down in [our documentation](https://docs.dataprep.ai/user_guide/clean/introduction.html). If you have further question on the usage, please feel free to post any question on this discussion page! Thank you very much and welcome your further contribution!
